### ISSUE TITLE
Nose shape is a bit strange

### ISSUE THREAD

**Issue by tigert** [c172p-team/c172p-detailed#216]
The model's engine cowling and the air inlets are a bit strange and too small.

Here are some reference pics if that might help: 
http://tigert.1g.fi/kuvat/Simulators/Flightgear/c172-details/

(Included are also some closeup shots of the wing tip and wheel pants, if someone wants to add those as an option.)

**Comment by gilbertohasnofb** [comment]
Probably related to https://github.com/Juanvvc/c172p-detailed/issues/6

**Comment by Juanvvc** [comment]
I was comparing our model with the reference photos, and our nose is definitely very, very strange. Our cone/spinner seems to be too small, and the holes for the air intake are definitely too small.

I tried to fix this a few days ago, but fixing the nose needs a few hours of work and, probably, a new UV map. I don't think we can fix this before the freeze.

**Comment by gilbertohasnofb** [comment]
> I tried to fix this a few days ago, but fixing the nose needs a few hours of work and, probably, a new UV map. I don't think we can fix this before the freeze.

Once this rush for the feature freeze is over, then we can work on this, Juan. Then I can also give you a hand by correcting the liveries if you change the UV map.

**Comment by onox** [comment]
The whole exterior 3D model could use a lot of improvements, so I'm gonna move this to 3.8 release.

**Comment by gilbertohasnofb** [comment]
Oh, this is totally a FG 3.8 release bug!

**Comment by thevirtualfer** [comment]
The original model was made with performance in mind. So, the modeler used few vertices and a lot of triangles and poles.

It works, but giving other shape to parts, will be a difficult task.

The better way to i see to deal with all these 3D issue, is retopologize the whole model, removing the triangles and the poles. Maybe a image could explain it better.

Look the original one. There are some triangles (to make curves) and some poles. But we don't see faceloops.

<a href='http://postimage.org/' target='_blank'><img src='http://s17.postimg.org/slkk0hwen/Screenshot_from_2015_08_20_09_28_13.png' border='0' alt="Screenshot from 2015 08 20 09 28 13" /></a><br /><a target='_blank' href='http://postimage.org/'>upload an image</a><br /><br />


We can retopo it. Making all faces quad, doing some loops, always with "change shape" in mind.

An example is the image below.

<a href='http://postimage.org/' target='_blank'><img src='http://s17.postimg.org/vgxn7d0en/Screenshot_from_2015_08_20_09_32_54.png' border='0' alt="Screenshot from 2015 08 20 09 32 54" /></a><br />

**Comment by onox** [comment]
If you want to retopologize the model, then this issue should not be closed?

**Comment by gilbertohasnofb** [comment]
> If you want to retopologize the model, then this issue should not be closed?

:+1:

**Comment by thevirtualfer** [comment]
Yes.
I will start tomorrow.

There are some issues related to this, that will be fixed by collateral effect.

**Comment by gilbertohasnofb** [comment]
@thevirtualfer cool! You should create a new branch to make your modifications, then make a pull request (PR). Once tested and accepted, we close all issues solved by the PR. Let me know if you need any help (our workflow changed quite a lot since the last time you have been around, but it changed for better :smile: )

And by the way, welcome back!

**Comment by Juanvvc** [comment]
@thevirtualfer, welcome back!

Please, take into account the internal organization of the main .blend file has changed a bit. Also, remember you where using a custom plugin (made by you!) to export the .blend file to .ac file. This plugin is now distributed with the project. Filename: io_scene_ac3d.zip Direct link: https://github.com/Juanvvc/c172p-detailed/blob/master/io_scene_ac3d.zip

**Comment by thevirtualfer** [comment]
Hi guys.

I am in trouble about the real dimension of the model.

I have 3 blueprints, but they are not orthographic and the proportions are different.

Just to know. Has any of you a good orthographic blueprint ?

Thanks @gilbertohasnofb, i will talk with you by e-mail, in portuguese to understand better.

@Juanvvc, i almost forget that script. Thanks. Will dig into blend file today to become tuned.

**Comment by gilbertohasnofb** [comment]
> i will talk with you by e-mail, in portuguese to understand better.

Sure, let me know if you have any problems with the current organization and I will try to help you the best I can!

**Comment by thevirtualfer** [comment]
In fact, @onox  gave me a very detailed tutorial. But i will ask you some things.

**Comment by dany93** [comment]
Hi, thevirtualfer,

I don't know if this is better than you already have, but I gave some suggestions [in the
US forum](http://forum.flightgear.org/viewtopic.php?f=4&t=25157&sid=5fa37b38b6e8fd0990ff5c2524435be0&p=254667#p254681)

**Comment by onox** [comment]
I'm not sure blueprints should always be used for modelling. They're often simplified or inaccurate. Better to use (many) photos.

**Comment by onox** [comment]
@thevirtualfer @Juanvvc FYI There's a FG user from the forums who has been improving the importer/exporter script. See https://github.com/NikolaiVChr/Blender-AC3D

**Comment by thevirtualfer** [comment]
@Juanvvc . Both lateral windows have this strange curve as corner. Is it correct ? I don't know if a real window has that shape. 

<a href='http://postimage.org/' target='_blank'><img src='http://s8.postimg.org/rd44bxrth/Garage_report_04.jpg' border='0' alt="Garage report 04" /></a><br />

**Comment by thevirtualfer** [comment]
Hi there, @onox.

I guess we are using exactly that script.

What we have done, was a simple hack to manage AC3D groups.

**Comment by thevirtualfer** [comment]
> I'm not sure blueprints should always be used for modelling. They're often simplified or inaccurate. Better to use (many) photos.

I tried. But photos have some perspective distortions. All photos i got, have some vanish point. I tried to set the camera's parameters to match the real cam. But never got a suitable result.

**Comment by gilbertohasnofb** [comment]
@wlbragg would you like to give a shot at the nose shape and window shapes? This one, together with the oil cap model we've been discussion, and we would have closed all external 3D model issues, I believe.

**Comment by wlbragg** [comment]
That was a pretty major fix, but I am glad we did it. That back window resolution has always bothered me. I managed to round it a bit more while I was doing the side. There is more that could be done to the bottom edge and bottom sides but pain to gain isn't really worth it, at least for me. I discovered several stay faces in both rear side window frames. We're talking maybe as many as 6-8 on each side, I removed them.
I also removed several edges from the forward side of the interior and exterior rear side windows and several faces from the top of the same windows.
Bottom line, even with the rounding of the rear window frame we probably had close to a wash in new vertices's from the whole procedure. Not that it matters, I don't think any of this is enough to have much effect on the overhead of the model.
I'm working on the oil door and vent next.
![fgfs-screen-001](https://user-images.githubusercontent.com/11069680/37245003-684c930c-2457-11e8-8517-1e8f13eb1d6d.png)

**Comment by gilbertohasnofb** [comment]
@wlbragg if the windows are too much trouble then I totally understand it. But if I understand you correctly, you seem to think that we tackled the nose but I don't think we did. This is how our nose looks like right now:

![fg](https://user-images.githubusercontent.com/5700795/37245138-b5096812-248b-11e8-929c-77ac2a92730c.jpg)

And now look at this post by @thevirtualfer here https://github.com/c172p-team/c172p-detailed/issues/216#issuecomment-133002957

As you see, our nose hasn't changed at all. The inlets are still just two billboards covering the hole, and the nose cone still does not fit into its hole. There were some arguments that our cone was too small and finally it should not be touching the fuselage. See:

![_dsc7938](https://user-images.githubusercontent.com/5700795/37245157-f9bf341e-248b-11e8-858a-7e0ae076bf2b.jpg)
![_dsc7992](https://user-images.githubusercontent.com/5700795/37245159-f9ee7cb0-248b-11e8-8d9a-fc1dbdca93cb.jpg)

So this is still an open issue even if we leave the windows as they are.

**Comment by gilbertohasnofb** [comment]
Summary of what needs to be done on this issue:

* inlets should be more outwards to the edges
* inlets should be carved a bit more before the air intake billboards
* cone needs to be enlarged, in particular its diameter
* fuselage hole where the cone mounts needs to be circular, not that terrible shape it has right now
- the cone should not be touching the fuselage, there should be a black cylinder connecting it (see last picture in the post above)

**Comment by wlbragg** [comment]
To avoid miscommunication here, I just finished the glass and the rear window frame, it's done.
My point is, you can always do more (like more resolution as in rounding the window frame), I'm not prepared to do that now. So as far as the side window frame is concerned as it relates to this issue, that part is finished even though it was an afterthought added to this "nose" issue, which is what I see after your last posts we are in agreement on.

Now, the "nose" issue, I was just getting ready to post that I am not going to try to tackle this. 
Here is my reasoning...
The stuff I have been doing is more like bodywork VS fabrication. After reading some of the other posts here with input from "professional" modelers, specifically https://github.com/c172p-team/c172p-detailed/issues/446 I don't have the skill set to do this justice without putting in more time I care to. 
However, I might consider a more minor fix.
Notice on the image below how the engine compartment is an afterthought, not even connected or created from the original fuselage mesh. I propose that I remove the separate engine compartment and extrude the actual cutouts in the fuselage to produce the same basic effect. The difference being it gets rid of an object and also eliminating the ugly seams. While I am at it I can try to edit it enough to make it look closer to what we had in mind.
The best fix is of course to start over and do it right from the beginning, but that is where a "professional" needs to be involved.
![nose3](https://user-images.githubusercontent.com/11069680/37245402-0739ea96-245d-11e8-98b6-fb6988f6ac1f.jpg)

**Comment by wlbragg** [comment]
P.S. I'll keep in mind your last bullet points while I am working on it.

**Comment by wlbragg** [comment]
One other reason of major importance here for my reluctance to get to far into this. If we start messing with the shape of the fuselage mesh you start affecting the UV layouts and the liveries. I know this are well enough to know this is going to be a major headache to re-UV-map this. That has been an ongoing theme for me during all of this work.

**Comment by gilbertohasnofb** [comment]
@wlbragg I was hoping that we could deform the current nose into a better shape so that it would not affect (too much) the UV-mapping, but I understand if you don't feel like tackling this. In that case, we should make sure to leave this issue opened after the new PR gets merged.

**Comment by wlbragg** [comment]
Here is my first pass in converting this. It is now all one piece and I reshaped almost everything. In Blender it looks like the openings are huge but in the sim, not so much.
![screenshot_2018-03-10_15-22-36](https://user-images.githubusercontent.com/11069680/37246853-f79cfcbc-2476-11e8-9766-618dc0a66ead.jpg)

![fgfs-screen-003](https://user-images.githubusercontent.com/11069680/37246836-860a3722-2476-11e8-8556-22124d9537c2.jpg)
![fgfs-screen-005](https://user-images.githubusercontent.com/11069680/37246838-8d2ff5e6-2476-11e8-8de8-f961c94d8f7e.jpg)


![fgfs-screen-004](https://user-images.githubusercontent.com/11069680/37246837-896a72d8-2476-11e8-9ec7-43a9d142481c.jpg)
![fgfs-screen-006](https://user-images.githubusercontent.com/11069680/37246842-952de348-2476-11e8-9f21-4af951eefaf6.jpg)

**Comment by wlbragg** [comment]
Either our prop hub and spinner are too small or I got the hight about right. The width on the other hand is a different story. Needs to extend quite a bit out towards the edges.

**Comment by wlbragg** [comment]
@gilbertohasnofb 
FYI: I already took a quick look at a couple liveries and either what I already pushed is messed up or the new stuff you haven't seen is. Just letting you know I am aware of the UV mapping issues and I will address that after I get the dimensions the way we want it.

**Comment by wlbragg** [comment]
@gilbertohasnofb 
You think these openings are big enough?
![fgfs-screen-007](https://user-images.githubusercontent.com/11069680/37247368-550c201e-247f-11e8-8ca9-7d8b14834eea.jpg)

**Comment by wlbragg** [comment]
@gilbertohasnofb 
This is about as good as I am going to get this. I am aware of several livery issues that I can't fix. Unfortunately I think your going to have to fix the livery themselves.
To point out a few things I noticed and why I can't fix it...
![liv](https://user-images.githubusercontent.com/11069680/37252094-5a1bf40c-24e9-11e8-8b45-827e8237b6e5.jpg)
These areas in red and green are the areas that need editing. The radi-box is now mapped appropriately but that part of the textures need reworked.
As far as the liveries are concerned when I fix one then another doesn't line up. I'm sure part of it is the fact that the mesh shape changed considerably. Even though it is going to take some effort on your part I think it is the only way we are going to get all the livery textures to line up, I can't do it through UV mapping alone.
The following pictures show liveries that I simply can't get to line up at the same time that the default lines up. If it would be easier for you to have me line up a particular livery and then you line up the rest by adjusting the texture itself, let me know which one you want to start with lined up. I didn't notice if more than one of the more detailed striped liveries were able to be lined up at the same time, I only know that to line up the default livery caused the others not to line up anymore. Looking at these images it looks like maybe a couple of them might line up at the same time and if so you would only have to fix the default texture. Even then though there are several others that need the top of the nose reworked. Also beware that some of the liveries were not correct in the nose area even before I worked on it. There are several that were messed up originally.
![fgfs-screen-005](https://user-images.githubusercontent.com/11069680/37252159-9aaf2c54-24ea-11e8-9292-e83236cf7344.png)
![fgfs-screen-004](https://user-images.githubusercontent.com/11069680/37252160-9c31c870-24ea-11e8-93c6-fa822ae36e76.png)
![fgfs-screen-003](https://user-images.githubusercontent.com/11069680/37252161-9dcabeb2-24ea-11e8-9fbf-fdb2aff0139d.png)
![fgfs-screen-002](https://user-images.githubusercontent.com/11069680/37252163-a25aa604-24ea-11e8-9461-656b2c8b7f42.png)

**Comment by gilbertohasnofb** [comment]
@wlbragg The nose is looking so much better already!

> Unfortunately I think your going to have to fix the livery themselves.

No problems, as soon as we 'freeze' the nose mesh I will fix those myself.

The new mesh looks much much better, but I do have some points to show to you:

1) I think the propeller blades should move a bit forward as well, they seem to be coming half from the spinner and half from behind it. Also, would it be possible to add a cylinder connecting the spinner to the fuselage, using this as reference?

![](https://user-images.githubusercontent.com/5700795/37245159-f9ee7cb0-248b-11e8-8d9a-fc1dbdca93cb.jpg)

![fgfs-screen-001](https://user-images.githubusercontent.com/5700795/37252807-f3528916-251f-11e8-8215-e98a78659a82.png)

2) I think the nose cone should be a bit more 'straight'. Would it be possible to select the middle circle of the cone and resize it's radius, so that there is not such a step? See ref above, and here is in the sim:

![fgfs-screen-004](https://user-images.githubusercontent.com/5700795/37252820-1a3c4c7e-2520-11e8-846c-8b4bcf14f312.png)

3) this is super nitpicky, but from a very specific angle I noticed that there is a strange uv-mapped triangle in the mesh behind the cone. If too much trouble just ignore it though:

![fgfs-screen-002](https://user-images.githubusercontent.com/5700795/37252825-3533b59e-2520-11e8-8990-d3e1f3fa1a8c.png)

**Comment by wlbragg** [comment]
>the propeller blades should move a bit forward as well, they seem to be coming half from the spinner and half from behind it. Also, would it be possible to add a cylinder 

I was planning on adding the cylinder last night but totally forgot about it and the prop moving forward, duuu, totally overlooked that detail as well. I'm glad your looking closely at this stuff. I suppose the blur disc should move sightly too.

**Comment by wlbragg** [comment]
OK, check this out. I made a new spinner from scratch so I could be sure it was consistently proportional, slightly rounded the tip instead of such a drastic point. Move the prop and blur, added a shaft and fixed the rotation animation so it doesn't wobble so much.

This is a part of the textures that needs some love. It's really ugly, it was flat broken because of the way it was UV mapped but should be a fairly simple fix now other than having to touch so many textures.
![radi](https://user-images.githubusercontent.com/11069680/37256676-00b9d5e4-252c-11e8-8b96-3b5bcee414df.jpg)

**Comment by wlbragg** [comment]
>from a very specific angle I noticed that there is a strange uv-mapped triangle in the mesh behind the cone

See if this was corrected, I never saw it but may have fixed it by making a new spinner.

**Comment by gilbertohasnofb** [comment]
@wlbragg Thanks again for yet more fixes! I really like the new spinner :smile: 

> See if this was corrected, I never saw it but may have fixed it by making a new spinner.

Not really, that was the mapping of the fuselage behind the spinner and so it is still there.

Also, this bit here still has a bit of a weird mesh:

![image](https://user-images.githubusercontent.com/5700795/37256764-21a9a4ac-2557-11e8-9fe2-e54303bd21d8.png)

As for the textures, thanks for pointing all this out to me. As soon as you say you are done altering the mesh on this PR I will fix all those textures.

**Comment by gilbertohasnofb** [comment]
One more thing: the side walls of the air intake should be in the colour of the livery, not in this mettalic colour. So all red in the image below should actually be fuselage colour:

![image](https://user-images.githubusercontent.com/5700795/37256802-b93ea948-2557-11e8-8e02-0e0add89aea6.png)

See:

![](https://user-images.githubusercontent.com/5700795/37245157-f9bf341e-248b-11e8-858a-7e0ae076bf2b.jpg)

So in that case, what about doing the following:
* map these walls to a new place in the `fuselage.png` which I will then add to each livery. There is some space on the left side of the spinner texture for instance (purple area in the shot below).
* the back side of the air intake should then be mapped to the whole air intake texture, red lines in the image below

![image](https://user-images.githubusercontent.com/5700795/37256830-12a68b2c-2558-11e8-86a1-b0e9ef3ceb1a.png)

**Comment by wlbragg** [comment]
OK, good idea because this is another area that needs texture work and this is the alternative. Your suggestion will make it much easier to map this stuff.
![map](https://user-images.githubusercontent.com/11069680/37257065-fa2d5f38-2531-11e8-9fe9-0e6e934124ec.jpg)

**Comment by wlbragg** [comment]
You want to post a scale texture here that I can use to map it too and when I am all finished you can add the textures to the branch/pr?

**Comment by wlbragg** [comment]
I suggest putting it her, might as well use this space closer to the radi-box.

![map2](https://user-images.githubusercontent.com/11069680/37257117-af516af8-2532-11e8-871f-f71688ee3c99.jpg)

**Comment by wlbragg** [comment]
![map2](https://user-images.githubusercontent.com/11069680/37257164-5673afee-2533-11e8-954b-744bc022dc53.jpg)

**Comment by wlbragg** [comment]
Obviously, put it wherever you want, I am still in Shuttle mode where my texture space got to be a premium.

**Comment by gilbertohasnofb** [comment]
You are right, there are better places for that texture. So try this, I used the position you last proposed just above:

![fuselage](https://user-images.githubusercontent.com/5700795/37257415-cc3727a6-2560-11e8-9490-385bf6563b9e.png)

**Comment by wlbragg** [comment]
OK, I think I'm done.
![map](https://user-images.githubusercontent.com/11069680/37257945-f8afd83c-253d-11e8-9eff-b7f17bdccd99.jpg)

**Comment by wlbragg** [comment]
The mesh around the air ducts is really a mess and chopped up, thus why you were seeing such bad lines in the light reflection. I tried to make some cuts to alleviate some of the issues but I don't know how effective it was. I think it is somewhat better, probably not much more can be done without major refactoring.

**Comment by gilbertohasnofb** [comment]
@wlbragg It is definitely looking much better! My only comments are:

1) that cylinder added to the spinner is not 'entering' the fuselage, there is some space in between, which is slightly noticeable from some angles. Would it be too much trouble to extended it a bit more?

2) this UV-mapping issue is still present, is this fixeable or is it not worth the trouble?:

![](https://user-images.githubusercontent.com/5700795/37252825-3533b59e-2520-11e8-8990-d3e1f3fa1a8c.png)

**Comment by wlbragg** [comment]
>spinner is not 'entering' the fuselage,

Oops, missed that, done.

>this UV-mapping issue is still present,

I'm not seeing anything that would cause this in the blend. Also I don't see it in the sim. How are you positioning in the sim to get that angle?
I recalculated the normals, see if that fixed it?
Also what is odd is that the entire face should be the color of the fuselage, not the dark gray. So the triangle is correct, the rest of it is what is wrong.

**Comment by gilbertohasnofb** [comment]
> Also I don't see it in the sim. How are you positioning in the sim to get that angle?

I am moving the cockpit camera around to inspect stuff, and this is the view from "inside" the spinner, though it is also visible from outside from a specific angle. The best livery to spot this is PT-IAO because of its colours, but the problem is with all liveries:

![fg-0](https://user-images.githubusercontent.com/5700795/37258237-b3cbdebc-256c-11e8-96d8-4bb76c640824.jpg)
![fg-1](https://user-images.githubusercontent.com/5700795/37258238-b3ee5118-256c-11e8-9057-6113af1404ff.jpg)

**Comment by wlbragg** [comment]
OK, I'm seeing it also, let me investigate and see exactly what this is.

**Comment by wlbragg** [comment]
I did a whole bunch more editing of the nose requiring a complete remapping of it. I'm in the process of mapping it now. :weary:

**Comment by wlbragg** [comment]
All yours! The default livery is all lined up (as far as I know), The rest of them are pretty darn close. Way closer than they were. The issue behind the spinner is fixed, that is what led to me doing even more to the nose. I am actually much happier with it, I think you'll notice a big difference.

**Comment by gilbertohasnofb** [comment]
Everything in the nose and spinner are looking wonderful, thank you so much! I did notice though that those white squares have been re-introduced at some point, on both sides of the cockpit:

![fgfs-screen-001](https://user-images.githubusercontent.com/5700795/37259260-ed44daea-257b-11e8-8675-2761eb2648b4.png)

I will wait for this and then we can freeze any UV-mapping in the nose/spinner. I don't think the UV-mapping will get affected if we end up adding the oil tank cap 3D model, right?

**Comment by wlbragg** [comment]
Oops, yep, forgot to move the faces for those when I remapped, let me do that, it'll take a second.

**Comment by wlbragg** [comment]
Done.

**Comment by gilbertohasnofb** [comment]
The UV-mapping of the sides of the fuselage close to the nose got messed up:

![fg-0](https://user-images.githubusercontent.com/5700795/37259495-04a7b34e-257f-11e8-9987-4ef9bf544569.jpg)
![fg-1](https://user-images.githubusercontent.com/5700795/37259496-04d0ed5e-257f-11e8-9967-c34db629d2d2.jpg)

I don't mean to be picky or anything, it's just that if I start to work on all liveries (26 in total) I need the UV-map to be as stable as possible, because every later modification causes an immense amount of work and pain and large git files :wink:

**Comment by wlbragg** [comment]
OK, the hinges must have been messed up for awhile now as I didn't touch those. I unfortunately screwed up the mapping below the windows when I tried to fix the light colored anomalies right below the windows. I had Blender in the wrong mode and it move all the connecting vertices, I didn't notice but I should have.
Try again!

**Comment by gilbertohasnofb** [comment]
Ok, just tested and I am still seeing some issues, sorry...

![fg-0](https://user-images.githubusercontent.com/5700795/37260291-53a9d90e-2589-11e8-95fa-dc13d88636f7.jpg)
![fg-1](https://user-images.githubusercontent.com/5700795/37260292-53c866ee-2589-11e8-89b1-8fb240ccf243.jpg)
![fg-2](https://user-images.githubusercontent.com/5700795/37260293-53e6f960-2589-11e8-9672-8f6c75e6a72f.jpg)


I am heading off for today, it's quite late here. So if you push the fixes I will work on the textures tomorrow, but can you please confirm to me that we are freezing all mesh and UV-mapping alterations to the nose section of the aircraft from now on on this PR? Also, can you confirm whether adding the oil cap as a 3D model later on will not affect the work I am about to start? I plan to start working on the textures by tomorrow.

**Comment by wlbragg** [comment]
Wow, the door handle in that livery may have been that way since I made them and we just never noticed. The problem is, in the default livery you have room to play with when lining them up. but not in other liveries. So even though I had them working for the default, not so for some of the detailed liveries. I wish whoever made the hinges would have just included them in the fuselage mesh VS a separate mesh, it would make this kind of stuff easier.

OK, I will fix the door and hinges and push tonight sometime.

The anomaly still under the window will have to be done in the textures as they are all slightly different at that location. We need some consistency at that location. The mapping needs to stay were it is and you can adjust the livery textures to match the default in that area.

No worries about the oil cap, I'll make it work and I doubt we will need a texture modification for that.

**Comment by gilbertohasnofb** [comment]
Sounds great, I will check it tomorrow then. Many thanks and good night!

**Comment by wlbragg** [comment]
I'll be gone all day tomorrow as well, After tonight I won't be back on it until late tomorrow afternoon or evening.

**Comment by wlbragg** [comment]
Good night!
